src/ObjectUtils/include/ObjectUtils/PdbDiaUtils.h

```diff
@@ -0,0 +1,352 @@
+          result_string += "HRESULT";
+          break;
+        case btChar16:  // 32
+          result_string += "char16_t";
+          break;
+        case btChar32:  // 33
+          result_string += "char32_t";
+          break;
+        case btChar8:  // 34
+          result_string += "char8_t";
+          break;
+        default:
+          ORBIT_ERROR("Unexpected base type with id \"%d\". Returning \"???\" as a type name.",
+                      base_type);
+          return "???";
+      }
+      result_string += parent_pointer_type_str;
+      return result_string;
+    }
+    case SymTagPointerType: {
+      IDiaSymbol* base_type = nullptr;
+      if (FAILED(type->get_type(&base_type)) || base_type == nullptr) {
+        return "???";
+      }
+      CComPtr<IDiaSymbol> base_type_comp = base_type;
+
+      std::string new_pointer_type_str;
+      if (BOOL is_reference; SUCCEEDED(type->get_reference(&is_reference)) && is_reference) {
+        new_pointer_type_str = "&";
+      } else if (BOOL is_rvalue_reference;
+                 SUCCEEDED(type->get_RValueReference(&is_rvalue_reference)) &&
+                 is_rvalue_reference) {
+        new_pointer_type_str = "&&";
+      } else if (BOOL is_ptr_to_member_function;
+                 SUCCEEDED(type->get_isPointerToMemberFunction(&is_ptr_to_member_function)) &&
+                 is_ptr_to_member_function) {
+        IDiaSymbol* class_parent = nullptr;
+        if (FAILED(type->get_classParent(&class_parent)) || class_parent == nullptr) {
+          return "???";
+        }
+        CComPtr<IDiaSymbol> class_parent_comp = class_parent;
+        std::string class_parent_str = PdbDiaTypeAsString(class_parent);
+        new_pointer_type_str = class_parent_str + "::*";
+      } else if (BOOL is_ptr_to_member_function;
+                 SUCCEEDED(type->get_isPointerToDataMember(&is_ptr_to_member_function)) &&
+                 is_ptr_to_member_function) {
+        IDiaSymbol* class_parent = nullptr;
+        if (FAILED(type->get_classParent(&class_parent)) || class_parent == nullptr) {
+          return "???";
+        }
+        CComPtr<IDiaSymbol> class_parent_comp = class_parent;
+        std::string class_parent_str = PdbDiaTypeAsString(class_parent);
+        new_pointer_type_str = class_parent_str + "::*";
+      } else {
+        new_pointer_type_str = "*";
+      }
+
+      if (BOOL is_const; SUCCEEDED(type->get_constType(&is_const)) && is_const) {
+        new_pointer_type_str += " const";
+      }
+      if (BOOL is_volatile; SUCCEEDED(type->get_volatileType(&is_volatile)) && is_volatile) {
+        new_pointer_type_str += " volatile";
+      }
+      if (BOOL is_unaligned; SUCCEEDED(type->get_unalignedType(&is_unaligned)) && is_unaligned) {
+        new_pointer_type_str += " __unaligned";
+      }
+      if (BOOL is_restricted;
+          SUCCEEDED(type->get_restrictedType(&is_restricted)) && is_restricted) {
+        new_pointer_type_str += " restricted";
+      }
```

Comment:
This piece of code appears twice in the PR. What about making it a free function?